api/src/main/java/org/openmrs/module/reporting/serializer/ReportingSerializer.java

```diff
@@ -85,6 +87,11 @@ public Object unmarshal(HierarchicalStreamReader reader, Object root) {
 	    xstream.registerConverter(new IndicatorConverter(mapper, converterLookup));
 
 		xstream.registerConverter(new ReportDefinitionConverter(mapper, converterLookup));
+
+		// Only setup XStreamSecurity only on versions that are after 2.7.0
+		if (new VersionComparator().compare(OpenmrsConstants.OPENMRS_VERSION, "2.7.0") >= 0) {
```

Comment:
@dkayiwa do we need this check now?

Reply:
This check was only set up XStream security on 2.7.0 and higher.

Reply:
If you do not have that check, you are going to end up with versions of openmrs core before 2.7 freezing on startup. Isn't that what you were fixing here? https://github.com/openmrs/openmrs-core/commit/f0910d4443bdaf2180662af55d33f36e925efda5

Or has your fix now confused you? 😊

Reply:
Because we did that fix we don't need this check now.

Reply:
How about those who are not yet ready to upgrade their platform versions?